src/osd/OSD.cc

```diff
@@ -4017,6 +4017,25 @@ PGRef OSD::handle_pg_create_info(const OSDMapRef& osdmap,
   PG::RecoveryCtx rctx = create_context();
 
   OSDMapRef startmap = get_map(info->epoch);
+
+  if (info->by_mon) {
+    int64_t pool_id = pgid.pgid.pool();
+    const pg_pool_t *pool = osdmap->get_pg_pool(pool_id);
+    if (!pool) {
```

Comment:
why doesn't this introduced a race when require_osd_release <= mimic, where the monitor won't create the pg again since it's unaware of the create message being discarded?

Reply:
The pool create and upgrade paths both ensure that we set CREATING if there are pgs to create and the OSDMap is mimic.  The code that sends the actual create messages doesn't know/care; we ensure the flag is set (or we are pre-mimic) if there are ever pgs queued for create.